agent/lib/src/api/model.rs

```diff
@@ -0,0 +1,187 @@
+use serde::{Deserialize, Serialize};
+
+
+/// Represents a file to be included in the workspace
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    /// Name of the file, paths relative to the workspace
+    pub filename: String,
+    /// Content of the file
+    pub content: String,
+}
+
+/// Identifies the type of the message
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    /// Status message to indicate that the agent is ready
+    Status,
+    /// Request message
+    Request,
+    /// Response message answering to a request message
+    Response,
+}
+
+/// Code to tell what the Request/Response message is about
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Code {
+    /// Represents a request to run the code or a response to such request
+    Run,
+    /// Indicates that your status is that you are ok and ready to communicate
+    Ok,
+}
+
+/// Represents a Status message
+#[derive(Deserialize, Serialize, Debug)]
+pub struct StatusMessage {
+    /// Type of the message
+    pub r#type: Type,
+    /// Code of the message
+    pub code: Code,
+}
+
+impl StatusMessage {
+    pub fn new(code: Code) -> StatusMessage {
+        StatusMessage {
+            // r#type is a reserved keyword in Rust, so we need to use the raw identifier syntax
+            r#type: Type::Status,
+            code
+        }
+    }
+}
+
+impl Default for StatusMessage {
+    fn default() -> Self {
+        Self::new(Code::Ok)
+    }
+}
+
+/// Serializes an Option<String> as a String by returning an empty string if the Option is None
+fn serialize_optionnal_string<S>(value: &Option<String>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    match value {
+        Some(v) => serializer.serialize_str(v),
+        None => serializer.serialize_str(""),
+    }
+}
+
+/// Represents the output of a step
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ResponseStep {
+    /// Command that was run
+    pub command: String,
+    /// Exit code of the command
+    pub result: i32,
+    /// Stdout of the command. If it is None, it will be serialized as an empty string 
+    /// to avoid api crashes
+    #[serde(serialize_with = "serialize_optionnal_string")] 
+    pub stdout: Option<String>,
+    /// Stderr of the command
+    pub stderr: String,
+    /// Whether the stdout should be returned or not
+    #[serde(alias = "enableOutput")]
+    pub enable_output: bool,
+}
+
+impl ResponseStep {
+    pub fn new(
+        command: String,
+        result: i32,
+        stdout: Option<String>,
+        stderr: String,
+        enable_output: bool,
```

Comment:
You should not need that if `stdout`is an `Option`

Reply:
I am not sure to understand what you mean here.

Reply:
You know if output is enabled or not by looking at `stdout`:

```rust
if let Some(r.stdout) {
    // output is enabled
} else {
   // output is disabled
}
```

So you no longer need the `enable_output` field.